WordPress/openverse
# Combine `VAudioDetails` and `VImageDetails` into `VMediaDetails` component

## Fixes



Related to #2731 by @fcoveram

## Description


To make the updates to the metadata section of the single result page easier, this PR replaces the separate `VAudioDetails` and `VImageDetails` components with a single `VMediaDetails` component. The metadata (source, provider, dimensions for images and sample rates etc for audio) are prepared in a new utility function in `metadata.ts`. This function creates a list of the title i18n keys, values and urls (if available) for the media metadata such as source/creator/file size/file format etc.

This PR also moves the tag filtering (to filter out empty tags) from the components to the `decodeMediaData` function that prepares each media item from the API response to be displayed on the frontend.

Some changes to the `metadata.ts` and `VMetadata` in this PR will be improved on in #2851. Here, the effort was made to extract the components without affecting the snapshots, as much as possible

## Testing Instructions

Run the app and see that the single result metadata looks correct, and works both on the audio and image pages.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
